components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/ApiStoreSdkGenerationManager.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    public Map<String, String> getSdkGenLanguages() {
+        return sdkGenLanguages;
+    }
+
+    /*
+        * This method generates the client side SDK for the API with API ID (apiID) and SDK language (language)
+        *
+        * @param apiId ID for the specific API
+        * @param language preferred language to generate the SDK
+        * @throws ApiStoreSdkGenerationException if failed to generate the SDK
+        * */
+    public String generateSdkForApi(String apiId, String language, String userName)
+            throws ApiStoreSdkGenerationException
+            , APIManagementException {
+
+        APIStore apiStore = APIManagerFactory.getInstance().getAPIConsumer(userName);
+        API api = apiStore.getAPIbyUUID(apiId);
+        if (api == null) {
+            String errorMessage = "Cannot find API for specified API ID";
+            throw new APIManagementException(errorMessage, ExceptionCodes.API_NOT_FOUND);
+        }
+        String apiName = api.getName();
+        String apiVersion = api.getVersion();
+        String swaggerDefinitionForApi = null;
+        try {
+            swaggerDefinitionForApi = apiStore.getApiSwaggerDefinition(apiId);
+        } catch (APIManagementException e) {
+            handleSdkGenException("Error retrieving swagger definition for API " + apiId + " from database.", e);
+        }
+        Swagger swaggerDoc = new SwaggerParser().parse(swaggerDefinitionForApi);
+        if (swaggerDoc == null) {
+            handleSdkGenException("Error while parsing retrieved swagger definition");
+        }
+        //Format the swagger definition as a string before writing to the file.
+        String formattedSwaggerDefinitionForSdk = Json.pretty(swaggerDoc);
+        Path tempSdkGenDir = null;
+        File swaggerDefJsonFile = null;
+        try {
+            //Create a temporary directory to store the API files
+            tempSdkGenDir = Files.createTempDirectory(apiName + "_" + language + "_" + apiVersion);
+
+            //Create a temporary file to store the swagger definition
+            swaggerDefJsonFile = Files.createTempFile(tempSdkGenDir,
+                    apiId + "_" + language,
+                    APIMgtConstants.APIFileUtilConstants.JSON_EXTENSION).toFile();
+        } catch (IOException e) {
+            handleSdkGenException("Error creating temporary directory or json file for swagger definition!", e);
+        }
+
+        String tempZipFilePath = "";
+        if (swaggerDefJsonFile.exists()) {
+
+            try (Writer swaggerFileWriter = new BufferedWriter(
+                    new OutputStreamWriter(
+                            new FileOutputStream(swaggerDefJsonFile.getAbsoluteFile()), "UTF-8"))) {
+
+                swaggerFileWriter.write(formattedSwaggerDefinitionForSdk);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Writing the swagger definition was sucessful to file " +
```

Comment:
Now we are not using is debug enable. Instead using singline debug statements with templates can we change this?